app/helpers/katello/katello_urls_helper.rb

```diff
@@ -51,11 +51,12 @@ def repository_url(content_path, _content_type = nil, schema = 'http')
               foreman_settings_url(schema)
             end
       content_path = content_path.sub(%r|^/|, '')
-      if @host.content_view && !@host.content_view.default?
-        content_path = [@host.content_view.label, content_path].join('/')
+      if @host.single_content_view_environment? && !@host.default_environment?
+        content_path = [@host.single_content_view.label, content_path].join('/')
       end
+      # TODO: add if @host.multi_content_view_environment?
       path = ::Katello::Glue::Pulp::Repos.repo_path_from_content_path(
-        @host.lifecycle_environment, content_path)
+        @host.single_lifecycle_environment, content_path) # TODO: figure out what to do for multi env
```

Comment:
@ianballou I got the test passing for single env, but need thoughts on how to adapt this for multi env..

Reply:
It appears that you can delete this entire method :)

Reply:
> It appears that you can delete this entire method :)

_Narrator:_ He could not, in fact, delete this entire method. https://bugzilla.redhat.com/show_bug.cgi?id=2269386